flink-table-store-format/src/main/java/org/apache/flink/table/store/format/parquet/ParquetInputFormatFactory.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.format.parquet;
+
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.connector.file.src.FileSourceSplit;
+import org.apache.flink.connector.file.src.reader.BulkFormat;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.store.utils.ReflectionUtils;
+import org.apache.flink.table.types.logical.RowType;
+
+import org.apache.hadoop.conf.Configuration;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collections;
+
+/** Factory to create parquet input format for different Flink versions. */
+public class ParquetInputFormatFactory {
+
+    public static BulkFormat<RowData, FileSourceSplit> create(
+            Configuration conf,
+            RowType producedRowType,
+            TypeInformation<RowData> producedTypeInfo,
+            boolean isUtcTimestamp) {
+        Class<?> formatClass = null;
+        try {
+            formatClass =
+                    Class.forName("org.apache.flink.formats.parquet.ParquetColumnarRowInputFormat");
+            return createFrom115(
```

Comment:
Could you please update the `createFrom115` to others? Otherwise if the future bumps the version to 1.16, should the method update to `createFrom116`?

Reply:
thanks for your reminds! I will update `createFrom115` to the better name. 
the second question, as official doc said that `Note: Table Store is only supported since Flink 1.14.`, we don't need to consider flink versions  before 1.14, until the latest version 1.16, `createPartitionedFormat()` has little changes, but `FlinkVersion` has changed frequently, for example in class name from 1.14 `MigrationVersion` to 1.15 `FlinkVersion`, and package name changed several times, introduce it would bring unexpected complexity and risks, WDYT?